Mage/src/main/java/mage/abilities/effects/common/SeekCardEffect.java

```diff
@@ -35,11 +54,24 @@ public SeekCardEffect copy() {
     @Override
     public boolean apply(Game game, Ability source) {
         Player controller = game.getPlayer(source.getControllerId());
-
         if (controller != null) {
-            return controller.seekCard(filter, source, game);
+            Set<Card> cards = controller.getLibrary()
+                    .getCards(game)
+                    .stream()
+                    .filter(card -> filter.match(card, getId(), source, game))
+                    .collect(Collectors.toSet());
+            Card card = RandomUtil.randomFromCollection(cards);
+            if (card == null) {
+                return false;
+            }
+            game.informPlayers(controller.getLogName() + " seeks a card from their library");
+            if (inZone == Zone.BATTLEFIELD) {
```

Comment:
New zone code must be part of `Player.seekCard` method (additional params), not here.

Reply:
I've decided to modify SeekCardEffect directly instead of using Player's seekCard() because of next problem. 
There are several cards that manipulate those cards that were found through the Seek. For example, [[Signature Spells] or [[Back-Alley Gardener]] or such a complex example like [[Plunderer's Prize
]]. 
So it seems that seekCard() method in PlayerImpl should return cards that it has found, for example as Cards interface. Would it be more appropriate way?

Reply:
Yes, changing the return type of that method to Card or Cards is the way to go (then when you need the boolean result, you check if null/empty)